java/arcs/android/storage/service/BindingContext.kt

```diff
@@ -74,81 +74,93 @@ class BindingContext(
+                        )
                     }
-                    resultCallback.onResult(null)
-                } catch (e: Throwable) {
-                    resultCallback.onResult(
-                        CrdtException("Exception occurred while awaiting idle", e).toProto()
-                            .toByteArray()
-                    )
                 }
             }
         }
     }
 
     @Suppress("UNCHECKED_CAST")
     override fun registerCallback(callback: IStorageServiceCallback): Int {
-        var callbackToken = 0
-        bindingContextStatisticsSink.traceTransaction("registerCallback") {
-            val proxyCallback = ProxyCallback<CrdtData, CrdtOperation, Any?> { message ->
-                // Asynchronously pass the message along to the callback. Use a supervisorScope here
-                // so that we catch any exceptions thrown within and re-throw on the same coroutine
-                // as the callback-caller.
-                supervisorScope {
-                    callback.onProxyMessage(message.toProto().toByteArray())
+        // TODO(b/162946893) - Make this oneway and add a result callback
+        val result = CompletableFuture<Int>()
+        scope.launch {
+            bindingContextStatisticsSink.traceTransaction("registerCallback") {
+                val proxyCallback = ProxyCallback<CrdtData, CrdtOperation, Any?> { message ->
+                    // Asynchronously pass the message along to the callback. Use a supervisorScope here
+                    // so that we catch any exceptions thrown within and re-throw on the same coroutine
+                    // as the callback-caller.
+                    supervisorScope {
+                        callback.onProxyMessage(message.toProto().toByteArray())
+                    }
                 }
-            }
 
-            callbackToken = runBlocking {
                 val token = (store() as ActiveStore<CrdtData, CrdtOperation, Any?>)
                     .on(proxyCallback)
 
                 // If the callback's binder dies, remove it from the callback collection.
                 callback.asBinder().linkToDeath({
-                    unregisterCallback(token)
-                }, 0)
+                                                    unregisterCallback(token)
+                                                }, 0)
 
-                token
+                result.complete(token)
             }
         }
-        return callbackToken
+        return result.get()
     }
 
     @Suppress("UNCHECKED_CAST")
     override fun sendProxyMessage(
         proxyMessage: ByteArray,
         resultCallback: IResultCallback
     ) {
-        bindingContextStatisticsSink.traceTransaction("sendProxyMessage") {
-            bindingContextStatisticsSink.measure(coroutineContext) {
-                resultCallback.takeIf { it.asBinder().isBinderAlive }?.onResult(null)
-
-                val actualMessage = proxyMessage.decodeProxyMessage()
+        scope.launch {
```

Comment:
Did you run runs_per_test=1000|10000 for tests related to StorageService as well as BindingContext?
I saw flakiness when I tried to use `scope.launch` on `bindingContextStatisticsSink.measure` couple weeks ago.
This might not stand true now, just want to double check it.

Reply:
Yes, I did a runs_per_test=1000 for the entire test suite, the flakiness was comparable to the flakes report I did on monday.